File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        result = []
        
        i = 0
        while i < len(intervals):
            low, high = intervals.pop(0)
            
            j = 0
            changed = False
            while j < len(intervals):
                if high < intervals[j][0] or intervals[j][1] < low:
                    j += 1
                else:    
                    this_low, this_high = intervals.pop(j)
                    low = min(low, this_low)
                    high = max(high, this_high)
                    changed = True
            
            if changed: 
                intervals = [[low, high]] + intervals
            else:    
                result.append([low, high])
            
        return result

File: 0056-merge-intervals/0056-merge-intervals.go
import (
    "sort"
)

func merge(intervals [][]int) [][]int {
    sort.Slice(intervals, func(i, j int) bool {
        return intervals[i][0] < intervals[j][0]
    })
    result := [][]int{ intervals[0] }
    
    for _, v := range intervals[1:] {
        if v[0] <= result[len(result)-1][1] {
            result[len(result)-1][1] = max(result[len(result)-1][1], v[1])
        } else {
            result = append(result, v)
        }
    }
    
    return result
}

func max(a, b int) int {
    if a < b {
        return b
    }
    return a
}